pkg/querier/querier.go

```diff
@@ -56,7 +56,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.QueryTimeout, "querier.query-timeout", 1*time.Minute, "Timeout when querying backends (ingesters or storage) during the execution of a query request")
 	f.DurationVar(&cfg.ExtraQueryDelay, "querier.extra-query-delay", 0, "Time to wait before sending more than the minimum successful query requests.")
 	f.DurationVar(&cfg.QueryIngestersWithin, "querier.query-ingesters-within", 0, "Maximum lookback beyond which queries are not sent to ingester. 0 means all queries are sent to ingester.")
-	f.IntVar(&cfg.MaxConcurrent, "querier.max-concurrent", 20, "The maximum number of concurrent queries.")
+	f.IntVar(&cfg.MaxConcurrent, "querier.max-concurrent", 20, `The maximum number of top-level LogQL queries that will execute at the same time, per querier process. It should be set to match the number of CPU cores available on the querier (or slightly less than the number of cores if you are in a shared environment). If using the query frontend, Then you should run enough queriers to match ('-querier.worker-parallelism' * 'number of query frontend replicas')/'-querier.max-concurrent'. Otherwise queries may queue in the query frontend, which will affect QoS. Alternatively, consider using '-querier.worker-match-max-concurrent' to force worker parallelism to match '-querier.max-concurrent'`)
```

Comment:
Can you make the prose here match the changes I put into lines 264-266?
Also, why did you change the double quotes delimiting the string into single back ticks? Looks wrong to me.

Reply:
👍